### DISCUSSION TITLE
NSGA-II and NSGA-III Settings

### DISCUSSION THREAD

**Discussion by alithearchitect** [Tunny-gh/Tunny_in_Open_Sea#197] [General]
Hi @hrntsm and Thank you so much for this wonderful plugin.
There's something I'd like to know about the settings. I've read the documentation, but I still need some clarification on how to configure it.
Specifically, I am trying to understand how to control the number of generations in the NSGA-II and NSGA-III algorithms. While I see an input for population size, I couldn't find a dedicated input for the number of generations. I initially thought the "Iteration" input in the Optimize tab was equivalent to the number of generations, but after some tests it seems this assumption was wrong.
Could you please provide guidance on how to specify the number of generations for these algorithms within the Tunny plugin? I want to make sure I'm using the tool correctly.

**Comment by hrntsm** [comment]
Hi @alithearchitect, Thank you for your interest.

Population size, as written, can be set from the following

<img width="286" alt="Screenshot 2023-09-08 at 10 48 18" src="https://github.com/hrntsm/Tunny/assets/23289252/267a3907-3ad3-46d7-99e3-1097b649ea0e">

Tunny does not have a section to set the generation directly, but it can be set using the Number of trials below.
For example, if the population size is 25 and the Number of trials is 100, optimization will be performed up to 4 generations (100/25 = 4).


<img width="290" alt="Screenshot 2023-09-08 at 10 26 06" src="https://github.com/hrntsm/Tunny/assets/23289252/7e0af6e3-8a1e-46fb-ab6d-90fc670bdb6b">

This is related to the fact that Tunny can finish the optimization once and then run it again using the results.
Although not visible in Tunny, each trial is assigned a generation number and information about the individual numbers within that generation. Therefore, if optimization is stopped and then continued, the information can be used to generate the next individual of the target generation.

**Reply by alithearchitect** [reply]
Thank you so much. It has now become clear to me.
